apps/test/unit/blockly/cdoSerializationHelpersTest.js

```diff
@@ -0,0 +1,179 @@
+import {expect} from '../../util/reconfiguredChai';
+import {
+  addPositionsToState,
+  getNewLocation,
+  getCursorYAdjustment,
+  isBlockLocationUnset,
+} from '@cdo/apps/blockly/addons/cdoSerializationHelpers';
+
+describe('addPositionsToState', () => {
+  it('should add x/y values from XML to JSON serialization', () => {
+    const xmlBlocks = [
+      {
+        blockly_block: {
+          id: 'block1',
+          type: 'when_run',
+        },
+        x: 20,
+        y: 20,
+      },
+      {
+        blockly_block: {
+          id: 'block2',
+          type: 'flappy_whenCollideGround',
+        },
+        x: 230,
+        y: 20,
+      },
+      {
+        blockly_block: {
+          id: 'block3',
+          type: 'flappy_whenClick',
+        },
+        x: 20,
+        y: 140,
+      },
+    ];
+
+    const blockIdMap = new Map([
+      ['block1', {type: 'when_run', id: 'block1', x: 0, y: 0}],
+      ['block2', {type: 'flappy_whenCollideGround', id: 'block2', x: 0, y: 0}],
+      ['block3', {type: 'flappy_whenClick', id: 'block3', x: 0, y: 0}],
+    ]);
+
+    addPositionsToState(xmlBlocks, blockIdMap);
+
+    expect(blockIdMap.get('block1')).to.deep.equal({
+      type: 'when_run',
+      id: 'block1',
+      x: 20,
+      y: 20,
+    });
+    expect(blockIdMap.get('block2')).to.deep.equal({
+      type: 'flappy_whenCollideGround',
+      id: 'block2',
+      x: 230,
+      y: 20,
+    });
+    expect(blockIdMap.get('block3')).to.deep.equal({
+      type: 'flappy_whenClick',
+      id: 'block3',
+      x: 20,
+      y: 140,
+    });
+  });
+});
+
+describe('getNewLocation', () => {
```

Comment:
These three tests do a nice job of each isolating one portion of the the logic in the function (one for RTL, one for blocks with SVG frames, one sort of "default" case).